src/include/interval_set.h

```diff
@@ -286,6 +286,41 @@ class interval_set {
       }
     }
   }
+
+  bool subset_size_sym(const interval_set &b) const {
+    typename decltype(m)::const_iterator pa = m.begin(),
+                                         pb = b.m.begin();
+
+    const typename decltype(m)::const_iterator a_end = m.end(),
+                                               b_end = b.m.end();
+
+    while (pa != a_end && pb != b_end) {
+
+      if (pb->first + pb->second <= pa->first)
+        { ++pb;  continue; }
```

Comment:
This will break if you give it input where pb runs through to the end.

Reply:
The `pb != b_end` bound is checked by the outer while loop. I did think about using an inner while loop here, but it didn't seem worth it since in only optimizes away one `pa != a_end` comparison.

Reply:
I'll go ahead and change it to a while loop with bounds check.

Reply:
Updated as follows:
```c++
      while (pb->first + pb->second <= pa->first) {
        ++pb;
        if (pb == b_end)
          return false;
      }
```